### ISSUE TITLE
Events to send to opposite hub

### ISSUE THREAD

**Issue by SynCap** [livestyle/sublime-text#7]
When typing in ST, LiveStyle send much of data so in browser occurs a lot of errors and in opposite, when typing in DevTools ST working hard. Often system may hangs up or be critically not responsive.
My idea - reduce intergange amount.
LS plugin in editor must wait for the caret will be moved to another **line**, but DevTools' plugin must wait for properties will been changed and updated.
+: switching to "less verbose mode" have to been changed by editor plugin settings.
++: on "save_file" command in editor, browser must reload the target css after a bit of time (might be possible to change latency in settings). This will let to get check changes made by native compiler.

Wish you success, and thank you for nice tool.

**Comment by sergeche** [comment]
> LS plugin in editor must wait for the caret will be moved to another line

No, it’s not an option: changing value in a property won’t be available for preview in this case

> DevTools' plugin must wait for properties will been changed and updated

DevTools doesn’t provide notifications about CSS property changes, in only notifies that Resource was updated

Can you show examples of errors?

**Comment by SynCap** [comment]
> changing value in a property won’t be available for preview in this case

Yes, with small css all looks amaizing, but for small css none additional tool is neccessary.

Preview while changing values in editor, especially complex properties, especially within large complex project, escpecially in LESS or SASS, produce huge garbage traffic and hangs up the both, as browser so editor.

I seem that will be better to finish the changes and then go to the another line. This will show the result MUCH FASTER thanks to the short queue of changes. Theremore reducing the garbage traffic (sending obviously erroneous data) will make the sycnronization much easier in bi-directional exchange mode.
For now LS causes the computer to do a lot of unneccessary work, as a result tool for easiing of work makes the process be painful.
So, I think that option in ST settings to switch conversation to less verbose mode is must have feature.

> only notifies that Resource was

Hmm. This sounds to be good. But updating the edior is going not correct sometimes, with partial updates for example, or insert updates into the wrong place.
May this occurs due the big update quees?

> Can you show examples of errors?

Some errors can be shown,, especialy wich reported in console of DevTools by LS. But for showing the errors in updates of browser view screencast is need to be recorded, but this becomes impossible because SYSTEM HANGS or so busy that any more to what is not the case. Even swithcing to the Photoshop in parallel becomes untrivial task! But I'll try to illustrate.

**Comment by sergeche** [comment]
I’m working on a new implementation of LiveStyle plugin for Chrome, it can reduce resource usage. The main problem here is not how often LiveStyle tries to recalculate data but how large is your stylesheet. If you’re editing a single file like 500 KB it may slow down your system for sure
